src/test/http_openai_handler_test.cpp

```diff
@@ -243,3 +245,252 @@ TEST_F(HttpOpenAIHandlerTest, GraphWithANameDoesNotExist) {
     auto status = handler->dispatchToProcessor("/v3/completions", requestBody, &response, comp, responseComponents, &writer);
     ASSERT_EQ(status, ovms::StatusCode::MEDIAPIPE_DEFINITION_NAME_MISSING);
 }
+
+class HttpOpenAIHandlerParsingTest : public ::testing::Test {
+protected:
+    rapidjson::Document doc;
+    std::shared_ptr<ov::genai::Tokenizer> tokenizer = std::make_shared<ov::genai::Tokenizer>(getGenericFullPathForSrcTest("/ovms/src/test/llm_testing/facebook/opt-125m"));
+};
+
+TEST_F(HttpOpenAIHandlerParsingTest, ParsingMessages) {
+    std::string json = R"({
+    "model": "llama",
+    "messages": [
+      {
+        "role": "user",
+        "content": [
+          {
+            "type": "text",
+            "text": "What is in this image?"
+          },
+          {
+            "type": "image_url",
+            "image_url": {
+              "url":  "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAAEElEQVR4nGLK27oAEAAA//8DYAHGgEvy5AAAAABJRU5ErkJggg=="
+            }
+          }
+        ]
+      }
+    ]
+  })";
+    doc.Parse(json.c_str());
+    ASSERT_FALSE(doc.HasParseError());
+    std::shared_ptr<ovms::OpenAIChatCompletionsHandler> apiHandler = std::make_shared<ovms::OpenAIChatCompletionsHandler>(doc, ovms::Endpoint::CHAT_COMPLETIONS, std::chrono::system_clock::now(), *tokenizer);
+    EXPECT_EQ(apiHandler->parseMessages(), absl::OkStatus());
+    std::vector<ov::Tensor> images = apiHandler->getImages();
+    ASSERT_EQ(images.size(), 1);
+    ov::Tensor image = images[0];
+    EXPECT_EQ(image.get_element_type(), ov::element::u8);
+    EXPECT_EQ(image.get_size(), 3);
+    std::vector<uint8_t> expectedBytes = {160, 181, 110};
+    for (size_t i = 0; i < image.get_size(); i++) {
+        EXPECT_EQ(expectedBytes[i], ((uint8_t*)image.data())[i]);
+    }
+    json = apiHandler->getProcessedJson();
+    EXPECT_EQ(json, std::string("{\"model\":\"llama\",\"messages\":[{\"role\":\"user\",\"content\":\"What is in this image?\"}]}"));
+}
+
+TEST_F(HttpOpenAIHandlerParsingTest, ParsingImageJpeg) {
```

Comment:
test name does not say that this is negative scenario - atleast one text message should exist (is it right?)